akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/Unpersistent.scala

```diff
@@ -118,14 +115,11 @@ private[akka] object Unpersistent {
       var shouldUnstash = false
       var shouldStop = false
 
-      def snapshotRequested(evt: Event): Boolean = {
-        val snapshotFromRetention = retention match {
-          case DisabledRetentionCriteria             => false
-          case s: SnapshotCountRetentionCriteriaImpl => s.snapshotWhen(sequenceNr)
-          case unexpected                            => throw new IllegalStateException(s"Unexpected retention criteria: $unexpected")
-        }
-
-        snapshotFromRetention || snapshotWhen(state, evt, sequenceNr)
+      def snapshotRequested(evt: Event): Boolean = retention match {
+        case DisabledRetentionCriteria                                 => false
+        case s: SnapshotPredicateCriteriaImpl[State, Event] @unchecked => s.predicate(state, evt, sequenceNr)
```

Comment:
Should we add a `SnapshotPredicateCriteriaImpl.snapshotWhen` as well?

Reply:
I'm not sure I understand the question.

Reply:
I think it is ok.